libcxx/test/libcxx/utilities/expected/expected.expected/transform_error.mandates.verify.cpp

```diff
@@ -13,8 +13,11 @@
 // UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
 
 // With clang-cl, some warnings have a 'which is a Microsoft extension' suffix
-// which break the tests.
-// XFAIL: msvc
+// which break the tests. But #102851 will turn it into an error, making the test pass.
```

Comment:
I think what we really want to do is add the diagnostics with a `expected-error 0-1` for the MSVC diagnostics and a comment that we can remove things in LLVM 22(?). As-is this will just result in the test being `UNSUPPORTED` indefinitely on MSVC.

Reply:
Could you elaborate it a bit please? I am not very familiar syntax of `expected-error` and the '0-1' feature wasn't present in lit's document. My concern is that modify the test to use `expected-error 0-1` would require modifying every single test case. And ideally, adding a comment at each line that is modified. It is not a clean solution 
IMO, the correct way to go forward is t overhaul libcxx test infrastructure to use ToT clang instead of release clang. It can solve a lot of inconsistency issues we have seen so far. But we all not that cannot be done in one day. Disabling this test and waiting for a clang roll on libcxx side is way going forward.

Reply:
If the diagnostic is always the same you can also make it `0-8`. It simply says we expect `0-1` errors. If there are different errors you can also just add a single `TODO` at the top of the file. Also, we should absolutely test with release versions of Clang, since we support them. We test _also_ with trunk.

Reply:
> If the diagnostic is always the same you can also make it `0-8`. It simply says we expect `0-1` errors. If there are different errors you can also just add a single `TODO` at the top of the file. Also, we should absolutely test with release versions of Clang, since we support them. We test _also_ with trunk.

Can I make the expect `0-1` errors only on msvc, what would the syntax looks like? For other platforms, test should through an error otherwise it should fail, we should guarantee that.

>  Also, we should absolutely test with release versions of Clang, since we support them. We test _also_ with trunk.
Not with Windows. As far as I can tell all windows bots for libcxx use release branch of clang. If they use ToT clang the test infra should have caught this issue.

Reply:
I don't think it's necessary to restrict it to a specific platform, but if you want to you should be able to just add `#ifdef _WIN32` around it. We're not testing clang here after all. The main thing we want to test here are the `static_assert`s.

Reply:
I have modified this change using `expected-error-re 0-1` and `expected-warning-re 0-1`. And added TODO at the top. Please take a look.